tests/exchange/test_withdrawal_handler.cairo

```diff
@@ -97,6 +153,111 @@ fn given_unexisting_key_when_cancel_withdrawal_then_fails() {
+        short_token: contract_address_const::<'short_token'>(),
+    };
+
+    data_store.set_market(key, 0, market);
+
+    start_mock_call(withdrawal_vault_address, 'record_transfer_in', 1);
+
+    let withdrawal_key = withdrawal_handler.create_withdrawal(account, params);
+
+    start_mock_call(data_store.contract_address, 'get_withdrawal', withdrawal);
+
+    // Key cleaning should be done in withdrawal_utils. We only check call here.
+    withdrawal_handler.cancel_withdrawal(withdrawal_key);
+}
+
+
+// This tests check withdrawal cancellation when market token amount is 0
+// It calls withdrawal_handler.cancel_withdrawal
+// The test expects the call to panic with the error 'empty withdrawal'.
+#[test]
+#[should_panic(expected: ('empty withdrawal amount',))]
+fn given_market_token_equals_zero_when_cancel_withdrawal_then_fails() {
+    let mut withdrawal = Withdrawal {
+        key: 0,
+        account: contract_address_const::<'account'>(),
+        receiver: contract_address_const::<0>(),
+        callback_contract: contract_address_const::<0>(),
+        ui_fee_receiver: contract_address_const::<0>(),
+        market: contract_address_const::<0>(),
+        long_token_swap_path: Default::default(),
+        short_token_swap_path: Default::default(),
+        market_token_amount: 0,
+        min_long_token_amount: 0,
+        min_short_token_amount: 0,
+        updated_at_block: 0,
+        execution_fee: 0,
+        callback_gas_limit: 0,
+    };
+
+    let (caller_address, data_store, event_emitter, withdrawal_handler, withdrawal_vault_address) =
+        setup();
+    start_prank(withdrawal_handler.contract_address, caller_address);
+
+    let account = contract_address_const::<'account'>();
+    let key = contract_address_const::<'market'>();
+    let mut params = create_withrawal_params(key);
+
+    let market = Market {
+        market_token: key,
+        index_token: contract_address_const::<'index_token'>(),
+        long_token: contract_address_const::<'long_token'>(),
+        short_token: contract_address_const::<'short_token'>(),
+    };
+
+    data_store.set_market(key, 0, market);
+
+    start_mock_call(withdrawal_vault_address, 'record_transfer_in', 1);
+
+    let withdrawal_key = withdrawal_handler.create_withdrawal(account, params);
+
+    stop_mock_call(withdrawal_vault_address, 'record_transfer_in');
+    start_mock_call(data_store.contract_address, 'get_withdrawal', withdrawal);
+
+    // Key cleaning should be done in withdrawal_utils. We only check call here.
+    withdrawal_handler.cancel_withdrawal(withdrawal_key);
+}
+
+// This tests check withdrawal execution when caller address doesn't meet controller role
+// It calls withdrawal_handler.execute_withdrawal
+// The test expects the call to panic with the error 'unauthorized_access'.
 #[test]
 #[should_panic(expected: ('unauthorized_access',))]
 fn given_caller_not_controller_when_execute_withdrawal_then_fails() {
```

Comment:
```suggestion
fn given_caller_not_keeper_when_execute_withdrawal_then_fails() {
```

Reply:
Hello, thx for the review. You mean assertion in the function name or create new tests to test data_store access succes/fail ?

Reply:
For this one it's just the function name but in the top comment I meant adding assertions to already existing functions such as `given_normal_conditions_when_create_withdrawal_then_works`.

Maybe you could check that `get_withdrawal` returns you the withdrawal you just created. And for `given_normal_conditions_when_cancel_withdrawal_then_works` check that the key returns an empty withdrawal from data_store

Reply:
Ok, i will improve tests. Thank you.

Reply:
Update done. I noticed two things :

- Withdrawal datas are created with key =0 : https://github.com/keep-starknet-strange/satoru/blob/main/src/withdrawal/withdrawal_utils.cairo#L151
- I have to increased gas to 200 to be abale to cancel a withdrawal : https://github.com/keep-starknet-strange/satoru/pull/549/files#diff-03af4f4679e003d13470c2f5065b5c5742d9346d598f9b7f46c07ce1d4090dd2R177

One last thing. Sempgrep security rules are failing. I will check later but i don't see what i introducethat is breaking this rule.

Reply:
Great catch ! You can add the right key to the withdrawal struct before storing it.

For semgrep it's not an issue you introduced. It's global on all PRs, we're investigating the issue

Reply:
Update done. Key added to withdrawal.